TST: test_as_euler_symmetric_axes tol with torch

While reviewing an array API PR locally on ARM (M3) Mac, I tried to get a baseline testsuite result on latest SciPy `main` and saw some failures that are reproducible via `spin test -t scipy/spatial/transform/tests -b torch`. Using `torch 2.8.0`.

A representative traceback is placed below the fold.

<details>

```
_______________________________________________________________________ test_as_euler_symmetric_axes[torch-True-seq_tuple5] _______________________________________________________________________

xp = <module 'scipy._lib.array_api_compat.torch' from '/Users/treddy/github_projects/scipy/build-install/usr/lib/python3.12/site-packages/scipy/_lib/array_api_compat/torch/__init__.py'>
seq_tuple = ('z', 'y', 'x'), intrinsic = True

    @make_xp_test_case(Rotation.from_euler, Rotation.as_euler)
    @pytest.mark.parametrize("seq_tuple", permutations("xyz"))
    @pytest.mark.parametrize("intrinsic", (False, True))
    def test_as_euler_symmetric_axes(xp, seq_tuple, intrinsic):
        # helper function for mean error tests
        def test_stats(error, mean_max, rms_max):
            mean = xp.mean(error, axis=0)
            std = xp.std(error, axis=0)
            rms = xp.hypot(mean, std)
            assert xp.all(xp.abs(mean) < mean_max)
            assert xp.all(rms < rms_max)
    
        rnd = np.random.RandomState(0)
        n = 1000
        angles = np.empty((n, 3))
        angles[:, 0] = rnd.uniform(low=-np.pi, high=np.pi, size=(n,))
        angles[:, 1] = rnd.uniform(low=0, high=np.pi, size=(n,))
        angles[:, 2] = rnd.uniform(low=-np.pi, high=np.pi, size=(n,))
        angles = xp.asarray(angles)
    
        # Rotation of the form A/B/A are rotation around symmetric axes
        seq = "".join([seq_tuple[0], seq_tuple[1], seq_tuple[0]])
        if intrinsic:
            seq = seq.upper()
        rotation = Rotation.from_euler(seq, angles)
        angles_quat = rotation.as_euler(seq)
>       xp_assert_close(angles, angles_quat, atol=0, rtol=1e-13)
E       AssertionError: Tensor-likes are not close!
E       
E       Mismatched elements: 1 / 3000 (0.0%)
E       Greatest absolute difference: 3.5171518475429764e-16 at index (705, 2) (up to 0 allowed)
E       Greatest relative difference: 1.0394814830591351e-13 at index (705, 2) (up to 1e-13 allowed)

angles     = tensor([[ 0.3067,  1.8626,  1.9573],
        [ 1.3521,  0.0316, -0.1503],
        [ 0.6457,  1.4949,  0.1455],
        ...,
        [ 2.7546,  1.5083, -1.1083],
        [-1.7050,  2.0228,  1.9701],
        [ 1.1130,  1.5764,  1.2403]])
angles_quat = tensor([[ 0.3067,  1.8626,  1.9573],
        [ 1.3521,  0.0316, -0.1503],
        [ 0.6457,  1.4949,  0.1455],
        ...,
        [ 2.7546,  1.5083, -1.1083],
        [-1.7050,  2.0228,  1.9701],
        [ 1.1130,  1.5764,  1.2403]])
intrinsic  = True
n          = 1000
rnd        = RandomState(MT19937) at 0x131AF8840
rotation   = Rotation.from_matrix(tensor([[[-0.1763,  0.3678,  0.9130],
                              [ 0.9157, -0.2789,  0.2892],
...                             [ 0.4164,  0.1481,  0.8970],
                              [-0.3245,  0.9459, -0.0056]]]))
seq        = 'ZYZ'
seq_tuple  = ('z', 'y', 'x')
test_stats = <function test_as_euler_symmetric_axes.<locals>.test_stats at 0x124267920>
xp         = <module 'scipy._lib.array_api_compat.torch' from '/Users/treddy/github_projects/scipy/build-install/usr/lib/python3.12/site-packages/scipy/_lib/array_api_compat/torch/__init__.py'>

scipy/spatial/transform/tests/test_rotation.py:974: AssertionError
```

</details>

A pretty simple patch seems to fix this up locally:

```diff
--- a/scipy/spatial/transform/tests/test_rotation.py
+++ b/scipy/spatial/transform/tests/test_rotation.py
@@ -971,7 +971,7 @@ def test_as_euler_symmetric_axes(xp, seq_tuple, intrinsic):
         seq = seq.upper()
     rotation = Rotation.from_euler(seq, angles)
     angles_quat = rotation.as_euler(seq)
-    xp_assert_close(angles, angles_quat, atol=0, rtol=1e-13)
+    xp_assert_close(angles, angles_quat, atol=0, rtol=1.1e-13)
```

Maybe we should just do that.

--- a/scipy/spatial/transform/tests/test_rotation.py
+++ b/scipy/spatial/transform/tests/test_rotation.py
@@ -971,7 +971,7 @@ def test_as_euler_symmetric_axes(xp, seq_tuple, intrinsic):
         seq = seq.upper()
     rotation = Rotation.from_euler(seq, angles)
     angles_quat = rotation.as_euler(seq)
-    xp_assert_close(angles, angles_quat, atol=0, rtol=1e-13)
+    xp_assert_close(angles, angles_quat, atol=0, rtol=1.1e-13)
     test_stats(angles_quat - angles, 1e-16, 1e-14)
 
 

